fem/bilininteg.hpp

```diff
@@ -266,6 +266,39 @@ class BilinearFormIntegrator : public NonlinearFormIntegrator
                                     Vector &flux, Vector *d_energy = NULL)
    { return 0.0; }
 
+   /** @brief For bilinear forms on element faces, specifies if the normal
+              derivatives are needed on the faces or just the face restriction.
+
+       @details if RequiresFaceNormalDerivatives() == true, then
+                AddMultPAFaceNormalDerivatives(...) should be invoked in place
+                of AddMultPA(...) and L2NormalDerivativeFaceRestriction should
+                be used to compute the normal derivatives. This is used for some
+                DG integrators, for example DGDiffusionIntegrator.
+
+       @returns whether normal derivatives appear in the bilinear form.
+   */
+   virtual bool RequiresFaceNormalDerivatives() const { return false; }
+
+   /// Method for partially assembled transposed action.
```

Comment:
Why "transpose" -- isn't this simply the action?

Reply:
Copy & paste bug. Fixed in 558fd4598.